horovod/runner/elastic/discovery.py

```diff
@@ -39,13 +52,48 @@ def get_event(self):
     def set_event(self):
         self._event.set()
 
+    def _in_cooldown_period(self, current_time):
+        return self._cooldown_period_end_ts > current_time
+
+
+    def _set_cooldown_period(self, current_time):
+        if self._cooldown_lower_limit == -1 or self._cooldown_upper_limit == -1:
+            return
+        self._blacklist_count += 1
+        def _exponential_backoff_time():
+            cooldown_delay = self._cooldown_lower_limit * (1 << self._blacklist_count)
+            + (random.uniform(0,1) * self._cooldown_lower_limit)
+            logging.debug(f"{self._blacklist_count}:{self._cooldown_period_end_ts} cooldown_delay: {cooldown_delay}")
+            return max(self._cooldown_lower_limit, min(self._cooldown_upper_limit, cooldown_delay))
+        cooldown_delta_seconds = _exponential_backoff_time()
```

Comment:
Why is this computation in an inner `def` anyway?

Reply:
I didn't feel I should expose the _exponential_backoff_time calculation to any other methods even in this class :). Are you suggesting I make it a regular private method?

Reply:
no, just inline the code into this `_set_cooldown_period` method